airbyte-integrations/connectors/source-chargebee/source_chargebee/streams.py

```diff
@@ -132,6 +117,78 @@ def read_records(
         # Always return an empty generator just in case no records were ever yielded
         yield from []
 
+
+class SemiIncrementalChargebeeStream(ChargebeeStream):
+    cursor_field = "updated_at"
+    # page_size = 1  # TODO Remove this field
+
+    def __init__(self, start_date: str):
+        # Convert `start_date` to timestamp(UTC).
+        self._start_date = pendulum.parse(start_date).int_timestamp if start_date else None
+
+    def get_starting_point(self, stream_state: Mapping[str, Any], item_id: str) -> str:
+        start_point = self._start_date
+
+        if stream_state and stream_state.get(item_id, {}).get(self.cursor_field):
+            start_point = max(start_point, stream_state[item_id][self.cursor_field])
+
+        return start_point
+
+    def parse_response(
+        self, list_result: ListResult, stream_slice: Mapping[str, Any] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Mapping]:
+        starting_point = self.get_starting_point(stream_state=stream_state, item_id=stream_slice["item_id"])
+
+        for message in list_result:
+            record = message._response[self.name]
+            if record[self.cursor_field] > starting_point:
```

Comment:
So you cant use "updated_at" field as filter parameter,  read all the  records and filter out in code? From my point of view its better to make this stream full refresh cause its over-complication for no reason. Was this approach approved by Eugene or Sheriff?

Reply:
Yes. We are already using such approach in `github` source.

Reply:
Btw it's not `over-complication for no reason`. The reason is very simple - output as few records as possible to the user. Because if a stream is working only in `full_refresh` mode then on each sync all existing in db data will be deleted and then it will be rewritten again + new records. It's much more time consuming.

For example, a user in db already has 1000 records for stream `some_name` and there are 10 new records in this stream that have to be added to the db.

In full resfesh mode:
1. 1000 old records will be deleted;
2. 1000 old records will be read;
2. 1000 old records will be added;
3. 10 old records will be read;
3. 10 old records will be added;

With such approach in which we do filtration in code we will have:
1. 1000 old records will be read;
2. 10 old records will be read;
3. 10 old records will be added;